proposals/NNNN-GetInteriorVehicleData.md

```diff
@@ -0,0 +1,96 @@
+-	If app1 has already subscribed data change notification for a moduleType, SDL still need to include parameter `subscribe=true` when forwarding requests from other apps, i.e. to re-subscribe the data again. There are extra messages and extra processing time.
+-	If app1 has not subscribed data change notification for a moduleType, SDL still need to include parameter `subscribe=false` when forwarding requests from other apps, i.e. to un-subscribe the data per app. There are extra messages and extra processing time.
+-	If app1 has already subscribed data change notification for a moduleType, and HMI returns the data to SDL in response,  SDL still forwards new GetInteriorVehicleData requests (from same app or different app regarding same moduleType) to HMI. However, SDL has the module data already and can keep the data up to date since SDL will get OnInteriorVehicleData notifications whenever there is any data change in the same moduleType. There are extra messages and extra processing time.
+-	It is not very clean. SDL still maintains the local status and make some filtering (cut off parameter `subscribe` in certain cases), why not always forward original message if we want to keep SDL simple.
+
+
+## Proposed solution
+
+The following sequence diagram shows the proposed behavior of sdl regarding GetInteriorVehicleData. 
+
+![../assets/proposals/nnnn-getInteriorVehicleData/GetInteriorVehicleData_new_subscribe.png](../assets/proposals/nnnn-getInteriorVehicleData/GetInteriorVehicleData_new_subscribe.png)
+
+SDL maintains the module (Type) subscription status as a whole. SDL need to subscribe to a module if there is at least one app subscribes to the module. SDL need to un-subscribe to a module if there is no app subscribes to the module. Or equivalently
+-	SDL shall only include parameter `subscribe=true` if SDL has not (equivalently there is no app) subscribed to the module data when forwarding GetInteriorVehicleData with `subscribe=true`.
+-	SDL shall only include parameter `subscribe=false` if it is the last app to unsubscribe to the module data when forwarding GetInteriorVehicleData with `subscribe=false`.
+
+
+SDL maintains a cache of module data only when there is at least an application subscribed to the module. 
+-	SDL creates a cache for the module when it receives a successful GetInteriorVehicleData response with `isSubscribed=true`. 
+-	SDL updates the cache (only the part that changes) when it receives an OnInteriorVehicleData notification. 
+-	SDL removes the cache for the module when SDL receives a successful GetInteriorVehicleData response with `isSubscribed=false`. 
+	
+
+SDL only forwards a GetInteriorVehicleData request to HMI if there is no cached data available for the requested module or it needs to un-subscribe to the module from HMI. Otherwise, SDL responds the resquest with the cached data and no need to forward to and wait a response from HMI.
```

Comment:
Please update to "unsubscribe" and change the last sentence to "Otherwise, SDL responds to the request with the cached data and there is no need to forward it and wait for a response from HMI."

Reply:
changed, thank you